packages/mask/src/extension/popups/pages/Friends/Home/index.tsx

```diff
@@ -1,39 +1,49 @@
 import { memo, useState, useMemo } from 'react'
 import { FriendsHomeUI } from './UI.js'
 import { useFriendsPaged } from '../../../hook/useFriends.js'
-import { EMPTY_LIST, NextIDPlatform } from '@masknet/shared-base'
+import { EMPTY_LIST } from '@masknet/shared-base'
 import { useTitle } from '../../../hook/useTitle.js'
 import { useI18N } from '../../../../../utils/i18n-next-ui.js'
-import { resolveNextIDPlatform, usePersonasFromNextID } from '@masknet/shared'
+import { resolveNextIDPlatform } from '@masknet/shared'
 import { useSearchValue } from '../../../hook/useSearchValue.js'
 import { useFriendsFromSearch } from '../../../hook/useFriendsFromSearch.js'
+import { useInfiniteQuery } from '@tanstack/react-query'
+import { NextIDProof } from '@masknet/web3-providers'
 
 const FriendsHome = memo(function FriendsHome() {
     const { t } = useI18N()
     useTitle(t('popups_encrypted_friends'))
 
-    const { data, fetchNextPage, isLoading, profilesArray } = useFriendsPaged()
-    const friends = useMemo(() => (data ? data.pages.flatMap((x) => x.friends) : EMPTY_LIST), [data])
-    const profiles = useMemo(() => (profilesArray ? profilesArray.pages.flat(1) : EMPTY_LIST), [profilesArray])
+    const { data, fetchNextPage, isLoading, refetch } = useFriendsPaged()
+    const friends = useMemo(() => data?.pages.flatMap((x) => x.friends) ?? EMPTY_LIST, [data])
     const [searchValue, setSearchValue] = useState('')
     const type = resolveNextIDPlatform(searchValue)
     const { loading: resolveLoading, value: keyword = '' } = useSearchValue(searchValue, type)
-    const { loading: searchLoading, value: searchResult } = usePersonasFromNextID(
-        keyword,
-        type ?? NextIDPlatform.NextID,
-        false,
+    const {
+        isLoading: searchLoading,
+        data: searchResultArray,
+        fetchNextPage: fetchNextSearchPage,
+    } = useInfiniteQuery(
+        ['search-personas', keyword, type],
+        async ({ pageParam }) => {
+            if (!type) return EMPTY_LIST
+            return await NextIDProof.queryExistedBindingByPlatform(type, keyword, pageParam ?? 1)
+        },
+        { enabled: !!keyword && !!type, getNextPageParam: (_, allPages) => allPages.length + 1 },
```

Comment:
When to finish the loading?
Should know when to finish the loading by return `undefined` in `getNextPageParam`